### ISSUE TITLE
API Respect SS_DATABASE_CHOOSE_NAME

### ISSUE THREAD

**Issue by robbieaverill** [silverstripe/cwp-installer#1]
**Migrated from GitLab**

cc @tractorcow 

I'm sick of all of my local projects using the same SS_cwp database when a better alternative is available.

This will make cwp developing better across multiple projects. :)

```diff
// find the database name from the environment file
if(defined('SS_DATABASE_NAME') && SS_DATABASE_NAME) {
	$database = SS_DATABASE_NAME;
- } else {
+ } elseif(!defined('SS_DATABASE_CHOOSE_NAME') || !SS_DATABASE_CHOOSE_NAME) {
	$database = 'SS_cwp';
}

```

**Comment by tractorcow** [comment]
Ah yes please! That would be great. :)

**Comment by NightJar** [comment]
With SS4 the configuration is both always from the environment, and always _before_ processing `_config`, so this is no longer an issue. There is no second chance to set a database name via the pseudo-environment (`SilverStripe\Core\Environment`) if one wasn't specified; Bootstrapping will have already failed with an error.

**Comment by robbieaverill** [comment]
This config is removed in #10

**Comment by robbieaverill** [comment]
Reopened for CWP 1.x resolution via #11. @NightJar should we merge up the branches to get the change back into the CWP 2.x version and switch it for env vars? I noticed it was removed in #10 for CWP 2.x

**Comment by NightJar** [comment]
Merging up would still result in the same end result; the `_config.php` would be removed as it is completely unnecessary in the `2.x` line. I don't think it's worth the effort, particularly for such a trivial change (which would then just be discarded again).

**Comment by robbieaverill** [comment]
Ok, good justification

**Comment by dhensby** [comment]
should this be closed?

**Comment by NightJar** [comment]
Wait for #11 since it was originally (and still is) labelled `affects/v3`

**Comment by robbieaverill** [comment]
Fixed in #11
